include/Architecture.hpp

```diff
@@ -364,20 +364,6 @@ class Architecture {
   void createDistanceTable();
   void createFidelityTable();
 
-  static double dijkstraNodeToCost(const Dijkstra::Node& node) {
-    // Dijkstra determines the minimal path cost from one physical qubit to
-    // another.  In the non-fidelity case we are only interested in swapping 2
-    // logical qubits next to each other. Therefore the last swap cost has to
-    // be ignored. That cost is stored in the field `node.prevCost` of each
-    // node.
-    if (node.containsCorrectEdge) {
-      return node.prevCost;
-    }
-    // in case the last edge is a back-edge, we will need to reverse the CNOT,
-    // executed on that edge
```

Comment:
Can we add comments along those lines to the newly introduced parts in the code so it becomes clearer what is going on? (E.g. why costs are no longer initialized with 0, but cost-reversal)

Reply:
I added some comments and doc strings that should now cover everything, I think.